### PR TITLE
K8SPSMDB-228: rename `psmdbCluster` to `clusterName` in `PerconaServerMongoDBBackup`

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPSMDB-228](https://jira.percona.com/browse/K8SPSMDB-228) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-228

### CODE REVIEW

**Review by defbin — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| upgrade | passed |
| non-voting | passed |
| upgrade-sharded | passed |
| pitr | failed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/9684d896d55f6e03324becbcb8b56a6d7b5e8413
image: `perconalab/percona-server-mongodb-operator:PR-816-9684d896`

File: pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go [hunk 1]
```diff
@@ -13,9 +13,9 @@ import (
 type PerconaServerMongoDBBackupSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
-	PSMDBCluster string              `json:"psmdbCluster,omitempty"`
-	StorageName  string              `json:"storageName,omitempty"`
-	Comperssion  pbm.CompressionType `json:"compressionType,omitempty"`
+	ClusterName string              `json:"clusterName,omitempty"`
```
Comment (delgod) [inline pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go:16]: I think we should keep backward compatibility.
let's check with @spron-in

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go (hunk 1 above)
Comment (hors) [reply pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go:16]: I think we should! @spron-in please check.

File: pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go [hunk 2]
```diff
@@ -13,9 +13,9 @@ import (
...
+	StorageName string              `json:"storageName,omitempty"`
+	Comperssion pbm.CompressionType `json:"compressionType,omitempty"`
```
Comment (dalbani) [inline pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go:18]: This merge request is maybe the opportunity to fix the typo in `Comperssion`?

File: pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go (hunk 1 above)
Comment (spron-in) [reply pkg/apis/psmdb/v1/perconaservermongodbbackup_types.go:16]: Yeah, we should. I left a comment in the JIRA ticket:

We are going to use `clusterName` everywhere

In 1.12 - 1.15 we are going to allow users to use `psmdbCluster` and `clusterName` in backup.yaml. But after 1.15 we are going to remove psmdbCluster completely.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| scheduled-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/6d255248d8ac319b451fcd736e52508e376de409
image: `perconalab/percona-server-mongodb-operator:PR-816-6d255248`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | failed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | failed |
| upgrade | passed |
| upgrade-sharded | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/e43daa28fae2a84cd286c05c38a01d67785a2d0f
image: `perconalab/percona-server-mongodb-operator:PR-816-e43daa28`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| limits | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| self-healing-chaos | passed |
| arbiter | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| data-sharded | passed |
| non-voting | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/e43daa28fae2a84cd286c05c38a01d67785a2d0f
image: `perconalab/percona-server-mongodb-operator:PR-816-e43daa28`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| init-deploy | passed |
| storage | passed |
| limits | passed |
| self-healing | passed |
| scaling | passed |
| self-healing-chaos | passed |
| security-context | passed |
| operator-self-healing | passed |
| rs-shard-migration | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/e43daa28fae2a84cd286c05c38a01d67785a2d0f
image: `perconalab/percona-server-mongodb-operator:PR-816-e43daa28`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/816/commits/80a26f563396b749e42298aabfceac9083f35963
image: `perconalab/percona-server-mongodb-operator:PR-816-80a26f56`

**Review by defbin — APPROVED** [review]

**Pull request merged**